### PR TITLE
Upgrade forbiddenapis to 2.6

### LINKED ISSUE #33759: Update forbiddenapis to v2.6
**Changelog: Version 2.6 (released 2018-09-17):**

> **New features:**
>   * Add Java 11 support by upgrading to ASM 6.2.1. This includes updated signatures files
>     and build system changes (shipped documentation is currently missing reference to
>     Java 11, sorry).
>   * The plugin was published on the [Gradle Plugin portal](https://plugins.gradle.org/plugin/de.thetaphi.forbiddenapis).
>     Due to a bug in plugin initialization, older versions may not fully work.
>     Thanks to Sebastian Davids and the Gradle team for help and pushing this forward!
>   * When signatures are using classes that are not found on classpath, the option to ignore
>     those warnings is no longer so noisy: It only lists all failed signatures separately
>     where methods/fields do not exit, but the missing classes are reported only with a
>     single line, thanks ghost,
>     Tim Allison, Dawid Weiss.
>   * Report more details about class that failed to load while scanning the classes
>     to be checked, thanks Trejkaz.
>   * Only print warning about Gradle daemon once.
> 
> **Bug fixes:**
>   * Gradle: Fix SourceSets added after plugin loading don't get tasks.
>   * Add a hack for the broken JavaVersion enum in Gradle. It returns version "1.9" and "1.10"
>     (which are invalid targetCompatibility versions) and this problem is only fixed after Java 11.
>     Before Java 11 we switch to plain "majorVersion" property, thanks Sebastian Davids.
>   * Gradle: Always apply JavaBasePlugin, otherwise startup may fail with new plugin DSL,
>     as initialization order is undefined, thanks Sebastian
>     Davids.
> 
> **Internals:**
>   * Refactor signatures file parsing and lookup of signatures (remove from Checker).
>   * Workaround for Java 1.6 (minimum version) builds with Maven Central TLS settings.
>   * Cleanup Gradle plugin initialization exception handling, check minimal Gradle version.

**Important:** Normally I would supply a PR, but due to the recent changes in Elasticsearch (use forbiddenapis CLI instead of plugin - why the hell???), I can't support it anymore (the CLI is also not actively maintained and stays for "simple use cases", it was never made for automated running from build scripts). Therefore, please do the update on your own. Just a few things:
- Forbiddenapis can handle target "11" (no version cap needed, or at least raise cap to 11)
- If you update (you should!) release branches that use the real Gradle plugin, please remove the commenting about the `server/java9` sourceset. The release fixed the bug in the Gradle plugin (later added sourcesets are not initialized with `forbiddenApisJava9` task). Due to many updates in the Gradle plugin, an update of release/maintenance branches should really be investigated (especially to check Java 9 code and compiled Groovy code).

Uwe

### PR DESCRIPTION
**Pull request by alpar-t**
- version bump 
- 11 is now supported, so no need to restrict it to 10 
- adjust to output change in forbiddenapis 

Closes #33759 ( w.r.t. the version upgrade, but the discussion there is welcome to continue )

### CODE REVIEW

File: buildSrc/src/main/java/org/elasticsearch/gradle/precommit/ThirdPartyAuditTask.java
Comment (uschindler) [inline buildSrc/src/main/java/org/elasticsearch/gradle/precommit/ThirdPartyAuditTask.java]: nich catch. Yes this message was changed due to the cleanup of missing class reporting.

Comment (atorok) [reply buildSrc/src/main/java/org/elasticsearch/gradle/precommit/ThirdPartyAuditTask.java]: ... a much appreciated change. Thanks for  that !

**Comment by atorok** [comment]
The failed CI job is due to the fact that we still run Gradle with java 8 there, and run into class not found errors when scanning the  java 9 classes which have references to classes introduced in 9.
This should not be a problem once #34179 is merged. Will have to merge this PR after that.

**Comment by uschindler** [comment]
I was just wondering. Because a while back you changed the minimum version to build Elasticsearch to Java 9 (or even later), but used the "-release" to compile the Java 8 classes.

**Comment by uschindler** [comment]
...BTW this was also a reason why I wondered that you fork anywhere in Elasticsearch's build. Since Java 9 it's perfectly possible to build 100% Java 8 compatible artifacts, as the "release" switch allows to actually compile against the older class signatures (real cross-compiling). IMHO, no forking is needed anymore, except for tests.

**Comment by atorok** [comment]
Yes, that is correct, we will only be working for tests. 
Third party audit still forks and runs the CLI version  but only because it needs to look at the output.
We could get rid of this if we had callbacks like `onMissingClass` and `onViolation` callbacks in the Gradle plugin.  The java version we use to fork  doesn't matter much, we make it the same as the target version only because it makes it easier to discover if something didn't go right, e.x. a java 9 specific class file was incorrectly extracted when targeting 8, we would get a class not found error, whereas it would silently continue if we were to run it with a never java.

**Comment by atorok** [comment]
> I was just wondering. Because a while back you changed the minimum version to build Elasticsearch to Java 9 (or even later), but used the "-release" to compile the Java 8 classes.

AFAIK we don't fully enforce that and CI was still running Gradle with java 8 using `-Dorg.gradle.java.home`. I'm in the process of cleaning that up.

**Comment by atorok** [comment]
@rjernst this is ready for review now and passes CI

**Review by rjernst — APPROVED** [review]
LGTM

**Pull request merged**
